stl/inc/xutility

```diff
@@ -4701,7 +4705,9 @@ _CONSTEXPR20 void fill(const _FwdIt _First, const _FwdIt _Last, const _Ty& _Val)
     _Adl_verify_range(_First, _Last);
     auto _UFirst      = _Get_unwrapped(_First);
     const auto _ULast = _Get_unwrapped(_Last);
-    if constexpr (_Fill_memset_is_safe<decltype(_UFirst), _Ty>) {
+    if constexpr (_Is_vb_iterator<_FwdIt>) {
+        return _Fill_vbool(_First, _Last, _Val);
+    } else if constexpr (_Fill_memset_is_safe<decltype(_UFirst), _Ty>) {

```

Comment:
This (and in `fill_n`) is the "dead code" metaprogramming that I mentioned in the other comment - if it activates, the classic loop on lines 4720-4722 occurs after a `return`. It should be sufficient to say: `if constexpr (detect only modifiable vector<bool>::iterator) { return cool new optimization; } else { all previous code }`.

Reply:
Done